lib/sublayer/providers/open_ai.rb

```diff
@@ -25,9 +25,9 @@ def self.call(prompt:, output_adapter:)
                   description: output_adapter.description,
                   parameters: {
                     type: "object",
-                    properties: OpenAI.format_properties(output_adapter)
+                    properties: output_adapter.format_properties
                   },
-                  required: [output_adapter.properties.select(&:required).map(&:name)]
```

Comment:
sidenote: the extra parentheses was actually a bug schema wise, though it did nothing negative to the prompt for gpt4